datatypes/src/primitives/time_interval.rs

```diff
@@ -61,6 +61,7 @@ impl TimeInterval {
     ///
     /// This constructor fails if `end` is before `start`
     ///
+    #[allow(clippy::trait_duplication_in_bounds)] // both `error::Error: From<A::Error> + From<B::Error>` is required
```

Comment:
do we actually need to support the case where A!=B?

Reply:
There are rare occasions where we used

```Rust
TimeInterval::new(0, TimeInstance::MAX)
```

or 

```Rust
TimeInterval::new(TimeInstance::MIN, 1)
```

or in the constructor of `STRectangle`.